api/envoy/extensions/filters/http/grpc_json_transcoder/v3/transcoder.proto

```diff
@@ -103,6 +103,17 @@ message GrpcJsonTranscoder {
     bool reject_binding_body_field_collisions = 3;
   }
 
+  message ReflectionConfig {
+    // Supplies the cluster name of the gRPC cluster to use for reflection.
+    // This will typically be the same gRPC cluster that Envoy is forwarding
+    // gRPC traffic to, but it does not need to be.
```

Comment:
Is the gRPC cluster that Envoy forwards to is the default value for this field? If so, please explicitly note so. If not, what is the default value?

Reply:
I don't think there is a default value for this field in the PR as written (beyond the protobuf default).  
This was an oversight on my part, thanks for catching!
Based on [this post](https://github.com/envoyproxy/envoy/issues/12074), it looks like Envoy can forward to multiple gRPC clusters.

So I have three questions.
1.Do you think we should specify a default value?
2. If we do specify a default value and there are multiple gRPC clusters configured, should we pick one arbitrarily or throw an error or log an error and do not initialize?
3. If we do not specify a default value, do you have thoughts on which of the following is preferable?
a) Logging an error and not initializing when no value is specified.
b) Throwing an exception and crashing Envoy when no value is specified.

Reply:
To follow up on the 3rd option you suggested, in the API we have ability to denote a string field as mandatory by setting its length ([example](https://github.com/envoyproxy/envoy/blob/0202852eb0d1423f86f72cb5ef735d459ffd9183/api/envoy/config/cluster/v3/cluster.proto#L807)).
The config will be rejected in case this is not populated.
If the value of this field needs to be declared somewhere else, please clarify the docs accordingly.

Reply:
How about using `envoy::config::core::v3::GrpcService`?    `cluster_name` is only for envoy_grpc.  GrpcService can support other grpc.

But on the other hand, this grpc server is the upstream server, normal grpc requests should be routed by envoy upstream.   So it is nature to use envoy_grpc.

Reply:
I'll defer to @adisuissa here, as the API shepherd.